File: HotelAPI/src/main/java/br/com/totvs/hotel/model/quarto/QuartoModel.java
package br.com.totvs.hotel.model.quarto;

import br.com.totvs.hotel.enumeration.quarto.CategoriaQuarto;
import br.com.totvs.hotel.enumeration.quarto.SituacaoQuarto;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "table_quarto")
public class QuartoModel {
    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false)
    private Integer numero;

    @Column(nullable = false)
    private String descricao;

    @Column(nullable = false)
    private Double precoHora;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CategoriaQuarto categoria;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private SituacaoQuarto situacao;

    @ElementCollection
    @CollectionTable(name = "table_quarto_imagens", joinColumns = @JoinColumn(name = "id_quarto"))
    @Column(name = "url", length = 65535)
    private List<String> imagens;

}

File: Hotel-Interface/accor/src/app/model/quarto.model.ts
export class Quarto {
    id: string;
    numero: string;
    descricao: string;
    categoria: string;
    situacao: string;
    precoHora: string;
    imagens: string[];

    constructor (id: string, numero: string, descricao: string, categoria: string, situacao: string, precoHora: string, imagens: string[]) {
        this.id = id;
        this.numero = numero;
        this.descricao = descricao;
        this.categoria = categoria;
        this.situacao = situacao;
        this.precoHora = precoHora;
        this.imagens = imagens;
    }
}
